pkg/client/client.go

```diff
@@ -50,10 +51,15 @@ func (c *Client) BuildConfigs(namespace string) BuildConfigInterface {
 }
 
 // BuildLogs provides a REST client for BuildLogs
-func (c *Client) BuildLogs(namespace string) BuildLogInterface {
+func (c *Client) BuildLogs(namespace string) BuildLogsInterface {
 	return newBuildLogs(c, namespace)
 }
 
+// BuildGenerate provides a REST client for Build generation
+func (c *Client) BuildGenerate(namespace string) BuildGenerateInterface {
```

Comment:
1) consistency is an admirable goal, but in this case i don't think it's worth it.  A client is a noun, thus BuildGenerator makes more sense than BuildGenerate.
2) I was actually going to suggest the endpoint be renamed to buildGenerator also, but since invoking the endpoint literally generates a build, it made sense to me that the endpoint be called generateBuild.